src/__tests__/hooks.test.js

```diff
@@ -827,33 +819,35 @@ describe.skip('hooks', () => {
         return value;
       };
 
-      executeAction = async () => {
+      const executeAction = async () => {
         const data = await handleAsyncAction();
         setState({ data });
       };
 
-      return <div>{state && state.data}</div>;
+      return (
+        <>
+          <button type="button" onClick={executeAction} />
+          <div>{state && state.data}</div>
+        </>
+      );
     };
 
-    // Get the first child since the page is wrapped with the WithTracking component.
-    const page = await mount(<Page />).childAt(0);
-    await page.instance().executeAction(); // TODO: this probably doesn't work (how well does Enzyme support hooks?)
-    // TODO: maybe have to just call executeAction(); here
-    executeAction();
+    const page = await mount(<Page />);
+    await page.find('button').simulate('click');
+    await new Promise(resolve => setTimeout(resolve));
```

Comment:
For some reason I needed to add this line to wait for the state to update for this test but not for the following one. My guess is that it has to do with the way the functions within the component are written—it's kind of baffling to me, but it seems to have to do with React's internals rather than our code and I don't think it has any actual implications for this PR—just a JS/React quirk.

Reply:
Maybe we have to use `act()` or something? https://reactjs.org/docs/test-utils.html#act

Reply:
Hmmm mayyyybe. I'll give it a shot, though usually jest yells at you to use it if it's necessary. The thing is, I've had to do this before with other tests to wait for the component to re-render with the new state, so that doesn't surprise me all that much—what does surprise me though is that the following test _doesn't_ seem to need to wait for a re-render. Anyway, I don't think it's something worth spending too much time on, but I will try using `act` to see if that helps.

Reply:
Ahhhh, we _did_ need `act`! But the error wasn't printing because we had previously mocked `console.error` and didn't unmock. Updating.

Reply:
Ah, good catch!